packages/ui/src/views/repo/components/branch-info-bar.tsx

```diff
@@ -15,13 +18,16 @@ interface BranchInfoBarProps {
 
 export const BranchInfoBar: FC<BranchInfoBarProps> = ({
   defaultBranchName = 'main',
+  repoId,
+  spaceId,
   useRepoBranchesStore,
+  selectedBranchTag,
   currentBranchDivergence
 }) => {
   const { behind, ahead } = currentBranchDivergence
-  const { repoId, spaceId, selectedBranchTag } = useRepoBranchesStore()
   const hasBehind = !!behind
   const hasAhead = !!ahead
+  const { selectedBranchTag: selectedBranchTagFromStore } = useRepoBranchesStore()
```

Comment:
`selectedBranchTag` can probably be passed down from the parent directly, and we should stop using the branchesStore